### PR TITLE
[OV20] More preprocessing tests 

### PR DESCRIPTION
**Pull request by nosovmik**
### Details:
- Use 'u8' element type in tests, as int8 template conversion has issues (to be fixed under separate tickets)
- Use 'float32' output for 'compare with reference' tests to avoid IE undesired conversions of output precisions
- Added GPU preprocessing tests skeleton (+ skip test, issue is raised)

### Tickets:
 - N/A

### CODE REVIEW

File: inference-engine/tests/ngraph_helpers/ngraph_functions/include/ngraph_functions/preprocess/preprocess_builders.hpp [hunk 1]
```diff
@@ -22,7 +22,13 @@ inline std::shared_ptr<Function> create_preprocess_1input(element::Type type,
...
     data1->set_friendly_name("input1");
     data1->output(0).get_tensor().set_names({"input1"});
-    auto res = std::make_shared<op::v0::Result>(data1);
+    std::shared_ptr<op::v0::Result> res;
+    if (type == element::f32) {
+        res = std::make_shared<op::v0::Result>(data1);
+    } else {
+        auto convert = std::make_shared<op::v0::Convert>(data1, element::f32);
+        res = std::make_shared<op::v0::Result>(convert);
+    }
```
[the remark this answers is not in the document]
Comment (nosovmik) [reply inference-engine/tests/ngraph_helpers/ngraph_functions/include/ngraph_functions/preprocess/preprocess_builders.hpp:31]: Yes, I guess it could be removed later, currently when output is 'u8' - I get a lot of mismatches like "expected=1", actual = "1.2", etc.

**Review by ilya-lavrenov — APPROVED** [review]

File: docs/template_plugin/tests/functional/subgraph_reference/preprocess.cpp [hunk 2]
```diff
@@ -101,23 +101,44 @@ static RefPreprocessParams scale_then_mean() {
...
+    res.inputs.emplace_back(Shape{1, 1, 2, 2}, element::i16, std::vector<int16_t>{2, 3, 4, 5});
+    res.expected.emplace_back(Shape{1, 1, 2, 2}, element::f32, std::vector<float>{0., 1., 1., 1.});
+    return res;
+}
+
 static RefPreprocessParams convert_element_type_and_scale() {
     RefPreprocessParams res("convert_element_type_and_scale");
     res.function = []() {
-        auto f = create_simple_function(element::i8, Shape{1, 3, 2, 2});
+        auto f = create_simple_function(element::u8, Shape{1, 3, 2, 2});
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/subgraph_reference/preprocess.cpp:126]: Why did you change `i8` to `u8`?

Comment (nosovmik) [reply docs/template_plugin/tests/functional/subgraph_reference/preprocess.cpp:126]: I stated in details, that reference implementation of 'convert to int8' has different rounding rules comparing to CPU/GPU, etc. Ticket for this is 66520. So I changed int8 to uint8 in preprocessing tests so that these are not dependent on this issue
